components/calendar/src/julian.rs

```diff
@@ -65,22 +82,17 @@ impl Calendar for Julian {
         _largest_unit: DateDurationUnit,
         _smallest_unit: DateDurationUnit,
     ) -> DateDuration<Self> {
-        DateDuration::new(
-            date1.0.year.0 - date2.0.year.0,
-            date1.0.year.0 - date2.0.year.0,
-            0,
-            u8::from(date1.0.day) as i32 - u8::from(date2.0.day) as i32,
-        )
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
     }
 
     /// The calendar-specific year represented by `date`
     fn year(&self, date: &Self::DateInner) -> types::Year {
-        date.0.year.into()
+        IsoYear(date.0.year).into()
     }
 
     /// The calendar-specific month represented by `date`
     fn month(&self, date: &Self::DateInner) -> types::Month {
-        date.0.month.into()
+        IsoMonth(date.0.month).into()
```

Comment:
This seems to be wrong, as `IsoMonth(u8)` is not public. Also, can't just return `date.0.month` as `u8` does not impl `for types::Month`

What is the suggested approach here?
Delete all usage of IsoMonth entirely and `Impl types::Month` for u8?

cc: @Manishearth

Reply:
> and `Impl types::Month` for u8?

This won't work, it's not a trait. Call `date.0.month.try_into().unwrap()` here.

Reply:
Actually, `types::Month` is a different type, you shouldn't need to be using IsoMonth here

https://unicode-org.github.io/icu4x-docs/doc/icu/calendar/types/struct.Month.html

Reply:
also for `day_of_month()` below it should be `types::DayOfMonth(date.0.day)`, probably.

Perhaps add a method to `ArithmeticDate` that does this for convenience